### PR TITLE
reside-205: Add deletion support

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds folder deletion support. This feels quite dangerous, in the wrong hands, but it will also be very useful. So to offset the damage, I've added a required `check` argument which would be the name of a file present in a folder, used as a way to verify that we do have the correct folder. Curious for your thoughts on this

At the same time, this has been updated to retire travis and use github actions

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
+# spud 0.1.6
+
+* Add a delete method to folders (reside-205)
+
 # spud 0.1.5
 
-* Rename package to spud (Sharepoint upload & delete) (reside-191)
+* Rename package to spud (Sharepoint upload & download) (reside-191)
```
Comment (r-ash) [inline NEWS.md:7]: oops - thanks!

File: R/sharepoint_folder.R [hunk 2]
```diff
@@ -93,6 +93,36 @@ sharepoint_folder <- R6::R6Class(
...
+    #'
+    #' @param path The path to delete. Use \code{NULL} to delete the current
+    #'   folder.
+    #'
+    #' @param check A file (not folder) that exists directly within
+    #'   \code{path}, used as a method to verify that you really do want
+    #'   to delete this folder (to prevent things like accidental deletion
+    #'   of the entire sharepoint, for example).
+    delete = function(path, check) {
+      if (!(check %in% self$files(path)$name)) {
```
Comment (r-ash) [inline R/sharepoint_folder.R:111]: I wonder if something like if you want to delete a folder it creates a confirm dialog where you have to write the name of the folder to confirm the delete. Kind of like github does when deleting a repo. Though I think this pattern probably fine, good that it goes into a recycle bin. Also maybe worth thinking if we can add a function which will reverse the delete action? Could move files back from the recycle bin - though maybe after an accidental delete it is easier to recover state through UI.

**Review by r-ash — COMMENTED** [review]
What's the use case for the delete here?

Did we ever manage to talk to microsoft about testing against a real server? Always worrying that the tests are basically all mocked. Maybe we can write some selenium tests which can use UI to check actions have actually happened? Though maybe more pain that it is worth.

File: R/sharepoint_folder.R (hunk 2 above)
Comment (richfitz) [reply R/sharepoint_folder.R:111]: I do need to do it in somewhat unsupervised code; use case is here: https://github.com/mrc-ide/roundtable/blob/prototype/R/rtm.R#L215-L216

**Review by r-ash — APPROVED** [review]

**Pull request merged**